mgmtsystem/hawkular.py

```diff
@@ -72,31 +74,66 @@ class Path(object):
 
     """
     def __init__(self, path):
-        self.paths = {'raw': path}
-        if self.raw:
-            raw_paths = re.split(r'(/\w+;)', self.raw)
-            if len(raw_paths) % 2 == 1:
-                del raw_paths[0]
-            for p_index in range(0, len(raw_paths), 2):
-                if PATH_NAME_MAPPING[raw_paths[p_index]] in self.paths:
-                    if isinstance(self.paths[PATH_NAME_MAPPING[raw_paths[p_index]]], list):
-                        self.paths[PATH_NAME_MAPPING[raw_paths[p_index]]] \
-                            .append(raw_paths[p_index + 1])
-                    else:
-                        v_list = [
-                            self.paths[PATH_NAME_MAPPING[raw_paths[p_index]]],
-                            raw_paths[p_index + 1]
-                        ]
-                        self.paths.update({PATH_NAME_MAPPING[raw_paths[p_index]]: v_list})
+        if path is None or len(path) == 0:
+            raise KeyError("CanonicalPath should not be None or empty!")
+        self._path_ids = []
+        r_paths = re.split(r'(/\w+;)', path)
+        if len(r_paths) % 2 == 1:
+            del r_paths[0]
```

Comment:
Under what conditions are the "extras" here, it should at least state in the docs that an extra is blindly dropped. If there shouldn't ever be one, we should raise an Exception